rule/rule.go

```diff
@@ -730,8 +730,11 @@ type attrValue struct {
 
 // NewRule creates a new, empty rule with the given kind and name.
 func NewRule(kind, name string) *Rule {
+	var call *bzl.CallExpr
 	kindIdent := &bzl.Ident{Name: kind}
```

Comment:
Thanks for the update, just one quick question: Why do we still need the `kindIdent`? Couldn't we set `Rule.kind` to the `DotExpr` or would that break other parts of Gazelle? Having two different AST representations in this structure makes me a bit worried.

Reply:
Actually I'm not sure to be honest, I don't know all Gazelle specifics, but afair `kind` needs to conform to one defined in `rule.KindInfo` map, e.g. for
```
var kinds = map[string]rule.KindInfo{
	"selects.config_setting_group": {
		...
	},
}
```
with `&bzl.Ident{Name: kind}` is matched to `selects.config_setting_group` string. If the same kind can be grabbed form the `DotExpr` then it will work as you're saying.

Reply:
`Rule.Kind()` string is made of `bzl.createDotExpr()` so it should work :)